common/element_handle.go

```diff
@@ -1521,29 +1373,64 @@ func (h *ElementHandle) Uncheck(opts goja.Value) {
 }
 
 func (h *ElementHandle) WaitForElementState(state string, opts goja.Value) {
-	rt := k6common.GetRuntime(h.ctx)
 	parsedOpts := NewElementHandleWaitForElementStateOptions(time.Duration(h.frame.manager.timeoutSettings.timeout()) * time.Second)
 	err := parsedOpts.Parse(h.ctx, opts)
 	if err != nil {
-		k6common.Throw(rt, fmt.Errorf("failed parsing options: %w", err))
+		k6Throw(h.ctx, "cannot parse element wait for state options: %w", err)
 	}
 	_, err = h.waitForElementState(h.ctx, []string{state}, parsedOpts.Timeout)
 	if err != nil {
-		k6common.Throw(rt, fmt.Errorf("error while waiting for state: %w", err))
+		k6Throw(h.ctx, "error while waiting for state: %w", err)
 	}
 }
 
 func (h *ElementHandle) WaitForSelector(selector string, opts goja.Value) api.ElementHandle {
-	rt := k6common.GetRuntime(h.ctx)
 	parsedOpts := NewFrameWaitForSelectorOptions(h.defaultTimeout())
 	if err := parsedOpts.Parse(h.ctx, opts); err != nil {
-		k6common.Throw(rt, fmt.Errorf("failed parsing options: %w", err))
+		k6Throw(h.ctx, "cannot parse element wait for selector options: %w", err)
 	}
 
 	handle, err := h.waitForSelector(h.ctx, selector, parsedOpts)
 	if err != nil {
-		k6common.Throw(rt, err)
+		k6Throw(h.ctx, "error while waiting for selector (%q): %w", selector, err)
 	}
 
 	return handle
 }
+
+// evalWithScript evaluates a given js function in the scope of this ElementHandle and allows to call the injected
+// script's functions.
+func (h *ElementHandle) evalWithScript(
+	ctx context.Context,
+	opts evaluateOptions, js string, args ...interface{},
+) (interface{}, error) {
+	// get the injected script's handle so that js function can call its functions.
+	script, err := h.execCtx.getInjectedScript(h.ctx)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get injected script: %w", err)
+	}
+	// passing `script` allows `js` to use `script`'s functions.
+	args = append([]interface{}{script}, args...)
+	return h.eval(ctx, opts, js, args...)
+}
+
+// eval evaluates a given js function in the scope of this ElementHandle.
+func (h *ElementHandle) eval(
+	ctx context.Context,
+	opts evaluateOptions, js string, args ...interface{},
+) (interface{}, error) {
+	// passing `h` makes it evaluate js code in the element handle's scope.
+	args = append([]interface{}{h}, args...)
+	// convert the arguments to goja values.
+	rt := k6common.GetRuntime(ctx)
+	gargs := make([]goja.Value, len(args))
+	for i, arg := range args {
+		gargs[i] = rt.ToValue(arg)
+	}
+	// call the javascript function.
+	result, err := h.execCtx.evaluate(ctx, opts, rt.ToValue(js), gargs...)
```

Comment:
Same here about the comments being redundant, though the first one about ```passing `h` ...``` is somewhat useful. :)

Also, should we rename `ExecutionContext.evaluate` to `eval` as well, to make it consistent?